src/core/test/system/setup.sol

```diff
@@ -133,11 +139,40 @@ contract TestSetup {
         lenderDeployer.deployAssessor();
         lenderDeployer.deployDistributor();
         lenderDeployer.deployJuniorOperator();
+
+        if (senior_) {
+            deploySenior(operator_);
+        }
+
         lenderDeployer.deploy();
 
         distributor = lenderDeployer.distributor();
         juniorOperator = lenderDeployer.juniorOperator();
         juniorERC20 = lenderDeployer.juniorERC20();
         junior = lenderDeployer.junior();
+        assessor = lenderDeployer.assessor();
+    }
+
+    function deploySenior(bytes32 operator_) public {
+        address sOperatorfab_;
+        address tranchefab_;
+
+        if (operator_ == "whitelist") {
+            sOperatorfab_ = address(new WhitelistOperatorFab());
+        } else if (operator_ == "allowance") {
+            sOperatorfab_ = address(new AllowanceOperatorFab());
+        }
```

Comment:
Currently, we always deploy both tranches with the same operator.

We could just pass the operatorFab instead of doing the if else again.